File: src/core/Selector.java
package core;

public interface Selector<T extends Generation<? extends Solution>> {
	/**
	 * "Процеживает" полученное поколение g, оставляя в нём одно или больше
	 * наилучших решений.
	 */
	
	public T keepTheBestSolutions(T g, T currentGeneration);
}

File: taboosearch/src/core/Selector.scala
package core
import scala.annotation.unchecked.uncheckedVariance

abstract class Selector[+S <: Solution] {
	type G <: Generation[S]
	def keepTheBestSolutions(generation : Generation[S] @uncheckedVariance, //Generation[S] @uncheckedVariance,
							 ctx : Context[Solution] @uncheckedVariance) : Generation[S];//Generation[S];
}